internal/ast/package.go

```diff
@@ -165,6 +167,47 @@ func (p *Package) GoArrayTraits(t *TypeReference) (string, error) {
 	}
 }
 
+// IsDeltaPackable indicates if a non-basic zserio type (struct, enum, subtype, ...)
+// can be delta-packed.
+func (p *Package) IsDeltaPackable(t *TypeReference) (bool, error) {
+	if t.IsBuiltin {
+		return p.BaseScope.Parent.IsDeltaPackable(t)
+	}
+
+	typeScope, err := p.GetImportedScope(t.Package)
+	if err != nil {
+		return false, err
+	}
+	typeSymbol, err := typeScope.GetSymbol(t.Name)
+	if err != nil {
+		return false, fmt.Errorf("%w: %s", ErrUnknownType, t.Name)
+	}
+
+	switch n := typeSymbol.Symbol.(type) {
+	case *Const:
+		return p.IsDeltaPackable(n.Type)
+	case *Subtype:
+		return p.IsDeltaPackable(n.Type)
+	case *InstantiateType:
+		return p.IsDeltaPackable(n.Type)
+	case *Struct:
+		// compound types have their own packed marshaling function
+		return false, nil
+	case *Enum:
+		return true, nil
+	case *Union:
+		// compound types have their own packed marshaling function
+		return false, nil
+	case *BitmaskType:
+		return true, nil
+	case *Choice:
+		// compound types have their own packed marshaling function
+		return false, nil
+	default:
+		return false, fmt.Errorf("%w: %s", ErrUnknownType, t.Name)
```

Comment:
```suggestion
	case *Const, *Subtype, *instantiateType:
		return p.IsDeltaPackable(n.Type)
	case *Struct, *Union, *Choice:
		// compound types have their own packed marshaling function
		return false, nil
	case *Enum, *BitmaskType:
		return true, nil
	default:
		return false, fmt.Errorf("%w: %s", ErrUnknownType, t.Name)
```

Reply:
Nice one - it doesn't work for `Const`, `Subtype `and `InstantiateType`, because the casted type is used in the case, I used it for all other cases.